### PR TITLE
feat(gh-action-crawling): add documentation

### PR DESCRIPTION
**Pull request by karanbirsingh**
See docs [here](https://github.com/karanbirsingh/accessibility-insights-action/tree/action-docs#-accessibility-insights-action) and [here](https://github.com/karanbirsingh/accessibility-insights-action/blob/action-docs/docs/usage.md#how-to-use-the-action).

This PR includes paired work with @lisli1 

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,20 +9,16 @@ Licensed under the MIT License.
 [![codecov](https://codecov.io/gh/microsoft/accessibility-insights-action/branch/main/graph/badge.svg)](https://codecov.io/gh/microsoft/accessibility-insights-action)
 [![Dependabot Status](https://api.dependabot.com/badges/status?host=github&repo=microsoft/accessibility-insights-action)](https://dependabot.com)
 
-Accessibility Insights Action helps integrate automated accessibility tests in GitHub workflow.
+_This action is still in early development & we don't recommend its usage in public production projects yet._
 
-**This is a pre-release version and therefore not production ready yet.**
```
Comment (ferBonnin) [inline README.md:14]: why are we removing the pre-release version comment?

Comment (karanbirsingh) [reply README.md:14]: We replaced it with this new phrasing, feel free to amend

_This action is still in early development & we don't recommend its usage in public production projects yet._

Comment (ferBonnin) [reply README.md:14]: got it, that phrase sounds great!

File: docs/usage.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v2
+
+            # Insert any jobs here required to build your website files
+
+            - name: Scan for A11y issues
```
Comment (lamaks) [inline docs/usage.md:48]:
```suggestion
            - name: Scan for accessibility issues
```

File: docs/usage.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                  name: accessibility-reports
+                  path: ${{ github.workspace }}/_accessibility-reports
+```
+
+### Scan local HTML files
+
+Provide the location of your built HTML files using `site-dir` and (optionally) `scan-url-relative-path`. The action will serve the site for you using `express`.
+
+```yml
+- name: Scan for A11y issues
```
Comment (lamaks) [inline docs/usage.md:68]:
```suggestion
- name: Scan for accessibility issues
```

File: docs/usage.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      site-dir: ${{ github.workspace }}/website/public
+      scan-url-relative-path: /index.html
+```
+
+### Scan a URL
+
+Provide the website URL. The URL should already be hosted - something like `http://localhost:12345/` or `https://example.com`.
+
+```yml
+- name: Scan for A11y issues
```
Comment (lamaks) [inline docs/usage.md:81]:
```suggestion
- name: Scan for accessibility issues
```

File: docs/usage.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+-   use `max-urls: 1` to turn off crawling
+-   include a list of additional URLs to scan (the crawler won't find pages that are unlinked from the base page)
+
+For `discovery-patterns`, `input-file`, and `input-urls`, note that these options expect resolved URLs. If you provide static HTML files via `site-dir`, you should also provide `localhost-port` so that you can anticipate the base URL of the file server (`http://localhost:localhost-port/`).
+
+Examples:
+
+```yml
+- name: Scan for A11y issues (with url)
```
Comment (lamaks) [inline docs/usage.md:103]:
```suggestion
- name: Scan for accessibility issues (with url)
```

File: docs/usage.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+```yml
+- name: Scan for accessibility issues (with url)
+  uses: microsoft/accessibility-insights-action@v1
+  with:
+      url: http://localhost:12345/
+      input-urls: http://localhost:12345/other-url http://localhost:12345/other-url2
+```
+
+```yml
+- name: Scan for A11y issues (with site-dir)
```
Comment (lamaks) [inline docs/usage.md:111]:
```suggestion
- name: Scan for accessibility issues (with site-dir)
```

File: docs/usage.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+You can choose to block pull requests if the action finds accessibility issues.
+
+1. Ensure the action is [triggered on each pull request](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#configuring-workflow-events).
+2. [Create a branch protection rule](https://docs.github.com/en/github/administering-a-repository/managing-a-branch-protection-rule#creating-a-branch-protection-rule) that requires the accessibility check to pass for a pull request to be merged.
+
+## Troubleshooting
+
+-   If the action didn't trigger as you expected, go to the ["on" section](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#on) of your yml file. Make sure any listed branch names are correct for your repository.
+-   If the action fails to complete, you can check the build logs for execution errors. Using the template above, these logs will be in the `Scan for A11y issues` step.
```
Comment (lamaks) [inline docs/usage.md:134]:
```suggestion
-   If the action fails to complete, you can check the build logs for execution errors. Using the template above, these logs will be in the `Scan for accessibility issues` step.
```

File: docs/usage.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+### Scan local HTML files
+
+Provide the location of your built HTML files using `site-dir` and (optionally) `scan-url-relative-path`. The action will serve the site for you using `express`.
+
+```yml
+- name: Scan for accessibility issues
+  uses: microsoft/accessibility-insights-action@v1
+  with:
+      repo-token: ${{ secrets.GITHUB_TOKEN }}
+      site-dir: ${{ github.workspace }}/website/public
```
Comment (lamaks) [inline docs/usage.md:72]:
```suggestion
      site-dir: ${{ github.workspace }}/website/root
```

File: docs/usage.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+  with:
+      url: http://localhost:12345/
+      input-urls: http://localhost:12345/other-url http://localhost:12345/other-url2
+```
+
+```yml
+- name: Scan for accessibility issues (with site-dir)
+  uses: microsoft/accessibility-insights-action@v1
+  with:
+      site-dir: ${{ github.workspace }}/website/public
```
Comment (lamaks) [inline docs/usage.md:114]:
```suggestion
      site-dir: ${{ github.workspace }}/website/root
```

**Review by lamaks — APPROVED** [review]

**Pull request merged**